File: app/src/main/java/com/alexkn/syntact/domain/SyntactApplication.java
package com.alexkn.syntact.domain;

import android.app.Application;

public class SyntactApplication extends Application {

    private SyntactComponent syntactComponent;

    @Override
    public void onCreate() {

        super.onCreate();
        syntactComponent = DaggerSyntactComponent.builder()
                .applicationModule(new ApplicationModule(this)).build();
        }

    public SyntactComponent getSyntactComponent() {

        return syntactComponent;
    }
}

File: app/src/main/java/dev/alexknittel/syntact/app/SyntactApplication.kt
package dev.alexknittel.syntact.app

import android.app.Application
import dev.alexknittel.syntact.BuildConfig
import dev.alexknittel.syntact.app.general.config.ApplicationComponent
import dev.alexknittel.syntact.app.general.config.ApplicationComponentProvider
import dev.alexknittel.syntact.app.general.config.DaggerApplicationComponent

open class SyntactApplication : Application(), ApplicationComponentProvider {

    override lateinit var applicationComponent: ApplicationComponent

    override fun onCreate() {
        System.setProperty("kotlinx.coroutines.debug", if (BuildConfig.DEBUG) "on" else "off")
        super.onCreate()
        applicationComponent = DaggerApplicationComponent.builder().applicationContext(this).build()
    }
}

val Any.TAG: String
    get() = javaClass.simpleName